test/common/http/http2/codec_impl_test.cc

```diff
@@ -3643,6 +3652,69 @@ TEST_P(Http2CodecImplTest, CanHandleMultipleBufferedDataProcessingOnAStream) {
   }
 }
 
+TEST_P(Http2CodecImplTest, ShouldTrackWhichStreamLeastRecentlyEncodedIfDeferProcessing) {
```

Comment:
I'm not sure I understand where in this test we're testing the new behavior. I'm sure it does so, but there's a lot of code and none of the comments seem to make it clear. Where is the key EXPECT?